src/Doctrine/DoctrineHelper.php

```diff
@@ -268,4 +307,27 @@ public function isKeyword(string $name): bool
 
         return $connection->getDatabasePlatform()->getReservedKeywordsList()->isKeyword($name);
     }
+
+    /**
+     * this method try to find the correct MappingDriver for the given namespace/class
+     * To determine which MappingDriver belongs to the class we check the prefixes configured in Doctrine and use the
+     * prefix that has the closest match to the given $namespace.
+     */
+    public function getMappingDriverForNamespace(string $namespace): ?MappingDriver
+    {
+        $lowestCharacterDiff = null;
+        $foundDriver = null;
+
+        foreach ($this->annotatedPrefixes as $key => $mappings) {
+            foreach ($mappings as [$prefix, $driver]) {
+                $diff = substr_compare($namespace, $prefix, 0);
```

Comment:
This works incorrect. I have installed package gesdinet/jwt-refresh-token-bundle wich has annotated entity. And I have next:
substr_compare('App\Entity\Test', 'App\Entity', 0);//5
substr_compare('App\Entity\Test', 'Gesdinet\JWTRefreshTokenBundle\Entity', 0);//-6
And AnnotationDriver returns instead of AttributeDriver

Reply:
Hi !
This behavior sounds logical, as the purpose of this part of the code is to return the driver used for a mappping.
So for Gesdinet\JWTRefreshTokenBundle\Entity, which is annotated, the fact that you have a AnnotationDriver is correct.
Am I missing something ?

Reply:
I have 2 prefixes (name spaces):
1) App\Entity - Attribute Driver
2) Gesdinet\JWTRefreshTokenBundle\Entity - Annotation Driver
I executed next command:
`php bin/console make:entity Test`
I expected that would be created App\Entity\Test entity with attributes, but I got - annotations!
I figured out that method `getMappingDriverForNamespace` returns incorrect Driver. Now this method returns the last driver of `$this->annotatedPrefixes` array, not the closest match.
I see 2 problems in mentioned block:
1) `$lowestCharacterDiff` always `null`. And if statement (row 325) always `true`
2) For entity `App\Entity\Test` closest match must be `App\Entity`. But
`substr_compare('App\Entity\Test', 'App\Entity', 0); ` results as `5`
`substr_compare('App\Entity\Test', 'Gesdinet\JWTRefreshTokenBundle\Entity', 0);` results as `-6`
-6 is less than 5, and this algorithm choosing incorrect driver
In my fork I using next variant of `if` block:
```php
if (null === $lowestCharacterDiff || ($diff < $lowestCharacterDiff && $diff > 0)) {
      $lowestCharacterDiff = $diff;
      $foundDriver = $driver;
}
```
I am not absolutely sure that it is correct, but it work for me.

Reply:
Ah ok, your first message wasn't that explicit.
I'm gonna check differently this part.

Reply:
This point has been fixed by @adlpz in #920